### DISCUSSION TITLE
Used By section is not displaying

### DISCUSSION THREAD

**Discussion by chamini2** [community/community#12101] [Actions] [answered]
> NOTE: first reported here: https://github.com/github/docs/issues/15750

Currently I am trying to get a public repo of a PyPI library to show the `Used by` section and I have had no luck on making it pick up the package.

- [PyPI package](https://pypi.org/project/fal/)
- [Repository](https://github.com/fal-ai/fal)
- [Other repo using said package](https://github.com/fal-ai/fal_dbt_examples/network/dependencies)
<img width="536" alt="image" src="https://user-images.githubusercontent.com/2745502/155229442-ad93fa0b-4c60-46b6-b46c-cdf03cb01225.png">

Note that the fal repository does not have a `setup.py` file but it is setup by poetry instead, maybe this has something to do?

**Comment by lseppala** [comment, marked as answer]
An explanation was provided in https://github.com/github/docs/issues/15750#issuecomment-1051349406, but repeating here for others: 

For the `fal` package, the reason there's no link to the repo in the Dependency Graph is because there is no `Source Code` URL set for the [package on PyPI](https://pypi.org/project/fal/). A button link to the source code repository will be rendered in the left sidebar of the package page when the URL is set. For instance, the `datadog_api_client` package [published in PyPI](https://pypi.org/project/datadog-api-client/) provides a `Source Code` URL to the GitHub repo. We are able to use this information to associate the package with the repository, even though the repo and package use different names.

We rely on information published with the public registries (like PyPI) to associate packages to their repositories. If the package in the registry doesn't provide this metadata, we typically don't try to assume where the source for this repository is (it could be anywhere!) 🙂 

The way to fix this would be to update the `fal` package in PyPI with a Source Code URL. The change will take time to take effect in our system (and it helps if there's a new version published in the registry that's used by a project, which triggers our system)

I hope this helps!

**Reply by chamini2** [reply]
Hey @lseppala, I have been waiting for a few days now and I don't see https://github.com/fal-ai/fal_dbt_examples/network/dependencies pointing to the right repository. Is there maybe something else we can do to help the system pick it up?

PD: Thought it was better to report here instead of docs repo.

**Reply by lseppala** [reply]
Hi @chamini2, thank you for reporting back to me!

The other idea I had that might help is publishing a new version _after_ updating the Source Code link on the package registry. A repository would also need to require this new version

It looks like you may have already done this. I see a new version (0.2.7) that was published 2022-02-28 and was [required in your project](https://github.com/fal-ai/fal_dbt_examples/commit/4112d171efdbeb5b69f2edbd0d3d1c7baba19b0e). 

I will dig into this today. Thank you for your patience and I'll let you know what I find! 🙂

**Reply by chamini2** [reply]
> It looks like you may have already done this. I see a new version (0.2.7) that was published 2022-02-28 and was [required in your project](https://github.com/fal-ai/fal_dbt_examples/commit/4112d171efdbeb5b69f2edbd0d3d1c7baba19b0e).

I did do that when I first published the source code, trying to _trigger_ the system check.

> I will dig into this today. Thank you for your patience and I'll let you know what I find! 🙂

Thank you!

**Reply by lseppala** [reply]
Hi @chamini2 

We've fixed the issue with `fal`, and it now appears in the Dependency Graph and Used By section  🎉 

![image](https://user-images.githubusercontent.com/5587177/156845545-a61cd8ed-9fee-4cb8-9ee9-db72724223ce.png)

![image](https://user-images.githubusercontent.com/5587177/156845812-46495709-13b4-45e1-b667-c6b9c6c4a15e.png)

You did everything correctly when updating the package in PyPI with a link. Our system failed to pick up the change, and will be fixed. Thank you for helping to address issues with our documentation and attribution system! 🙌

**Reply by chamini2** [reply]
This is great! Thank you for your help @lseppala! Hope this ended up improving the system!

**Comment by jxnu-liguobin** [comment]
Hi, @chamini2 , I would like to know why it(`org.bitlap:smt-csv-core_2.13) is not linked?  https://github.com/bitlap/bitlap/network/dependencies

**Reply by chamini2** [reply]
I really do not know. I think you can ask in a new discussion here? Or maybe @lseppala could take a look.
